packages/flutter_driver/lib/src/driver/driver.dart

```diff
@@ -877,9 +877,9 @@ class FlutterDriver {
   /// [getFlagList]: https://github.com/dart-lang/sdk/blob/master/runtime/vm/service/service.md#getflaglist
   Future<List<Map<String, dynamic>>> getVmFlags() async {
     await _restorePeerConnectionIfNeeded();
-    final Map<String, dynamic> result = await _peer.sendRequest('getFlagList');
+    final Map<String, dynamic> result = await _peer.sendRequest('getFlagList') as Map<String, dynamic>;
     return result != null
-        ? result['flags'].cast<Map<String,dynamic>>()
+        ? (result['flags'] as List<dynamic>).cast<Map<String,dynamic>>()
```

Comment:
The double cast is strange...

Reply:
...but necessary. Same comment as above.

Reply:
I ran into this same thing when deserializing responses from Gold. Double casting was the only way in that case too.